backend/regalloc/regalloc_rewrite.ml

```diff
@@ -37,16 +37,81 @@ type direction =
   | Load_after_list of Cfg.basic_instruction_list
   | Store_before_list of Cfg.basic_instruction_list
 
+(* Applies an optimization on the CFG outputted by [rewrite_gen] having one
+   temporary per variable per block rather than one per use of the variable,
+   reducing the number of spills and reloads needed for variables used multiple
+   times in a block. It iterates over each block and builds a substitution from
+   the first used temporary for each variable to all the other temporaries used
+   later for that variable, deleting now redundant reload/spill instructions
+   along the way. This currently does not support the case where spilled nodes
```

Comment:
Does not supported imply that the instruction-level spill is untouched for instructions that can use a spilled operand?

Reply:
This optimization is unsound when spilled nodes are used directly in instructions (if allowed by the ISA) without a new temporary being created (hence stack operands are not used in `rewrite_gen` if this optimization is enabled).

I have updated the comment.